src/OpenLoco/src/World/Station.cpp

```diff
@@ -1055,6 +1055,50 @@ namespace OpenLoco
         }
     }
 
+    // 0x0048F43A
+    void removeTileFromStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        auto findPos = pos;
+        findPos.z |= rotation;
+
+        // Find tile to remove
+        auto foundTilePos = std::find(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
+        if (foundTilePos == std::end(station->stationTiles))
+        {
+            // Bug mitigation: ensure stationTileSize does not exceed the actual array length
+            station->stationTileSize = std::clamp<int16_t>(station->stationTileSize - 1, 0, static_cast<uint16_t>(std::size(station->stationTiles)));
+            return;
+        }
+
+        // Remove tile by moving the remaining tiles over the one to remove
+        // NB: erasing is handled by StationManager::zeroUnused; not calling std::erase due to type mismatches
+        std::rotate(foundTilePos, foundTilePos + 1, std::end(station->stationTiles));
```

Comment:
while this is correct it won't match vanilla so will break replays.

Reply:
Argh, of course. Should we just revert the latest commit in the PR  then?

Reply:
tbh i don't really mind making this change it all gets cleanedup by zeroUnused. I've run this branch (after making the  ghost fix) and it matches replays but thats probably because zeroUnused ended up getting called at some point.

Reply:
Oh good. If it's not breaking replays, then I think this is the cleaner implementation compared to the previous one. Let's leave it then.